bokehjs/src/lib/models/math_text/math_text.ts

```diff
@@ -0,0 +1,226 @@
+    const face = v.text_font.get_value()
+
+    const {font_size_scale} = this
+    if (font_size_scale != 1.0) {
+      const match = size.match(/^\s*(\d+(\.\d+)?)(\w+)\s*$/)
+      if (match != null) {
+        const [, value,, unit] = match
+        const number = Number(value)
+        if (!isNaN(number))
+          size = `${number*font_size_scale}${unit}`
+      }
+    }
+
+    const font = `${style} ${size} ${face}`
+    this.font = font
+    this.color = color2css(color, alpha)
+  }
+
+  /**
+   * Calculates position of element after considering
+   * anchor and dimensions
+   */
+  protected _computed_position(): {x: number, y: number} {
+    const {width, height} = this
+    const {sx, sy, x_anchor="left", y_anchor="center"} = this.position
+
+    const x = sx - (() => {
+      if (isNumber(x_anchor))
+        return x_anchor*width
+      else {
+        switch (x_anchor) {
+          case "left": return 0
+          case "center": return 0.5*width
+          case "right": return width
+        }
+      }
+    })()
+
+    const y = sy - (() => {
+      if (isNumber(y_anchor))
+        return y_anchor*height
+      else {
+        switch (y_anchor) {
+          case "top": return 0
+          case "center": return 0.5*height
+          case "bottom": return height
+          case "baseline": return 0.5*height
+        }
+      }
+    })()
+
+    return {x, y}
+  }
+
+  /**
+   * Uses the width, height and given angle to calculate the size
+  */
+  size(): Size {
+    const {width, height} = this.getDimensions()
+    const {angle} = this
+
+    if (!angle)
+      return {width, height}
+    else {
+      const c = Math.cos(Math.abs(angle))
+      const s = Math.sin(Math.abs(angle))
+
+      return {
+        width: Math.abs(width*c + height*s),
+        height: Math.abs(width*s + height*c),
+      }
+    }
+  }
+
+  /**
+   * Get width and height from SVGElement
+   */
+  private getDimensions(): Size {
+    if (typeof MathJax === "undefined") return {width: 0, height: 0}
+
+    const mathjax_element = MathJax.tex2svg(this.model.text)
+
+    document.body.appendChild(mathjax_element)
+
+    const svg_element = mathjax_element.children[0] as SVGElement
+    svg_element.setAttribute("font", this.font)
+
+    const {width, height} = svg_element.getBoundingClientRect()
+
+    mathjax_element.remove()
+
+    return {width, height}
+  }
+
+  /**
+   * Calls ImageLoader after rendering the text into a SVG with MathJax.
+   * Starts Mathjax script loading if its not present in the global context.
+   */
+  load_image(): void {
+    if (!document.getElementById("bokeh_mathjax_script") && typeof MathJax === "undefined") {
+      const script = document.createElement("script")
+      script.id = "bokeh_mathjax_script"
+      script.src = "https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"
+      document.head.appendChild(script)
+    }
+
+    if (typeof MathJax === "undefined" || this.image_is_loading) return
```

Comment:
So, what happens when MathJax fails to load? There needs to be a fallback mechanism. An obvious minimum would be just to render raw math text. We may also want to log (using bokehjs' logger) that an error occurred, instead of relying on the browser default error messaging, which may not be that useful for our users. And btw. all of this needs to be tested (in context of earlier discussion of having a local way of serving mathjax for testing).

Reply:
Also, this needs to be moved out to its own method, or maybe even a new resources loader class, especially if we want to support retries, timeouts, etc. Some inspiration can sourced from `ImageLoader` (some!; that class is ideal).

Reply:
While we are at this, we may want to try using ESM instead of scripts that modify the global context. MathJax doesn't provide bundled ES module on its own, but internally they use ESM and, based on a rudimentary research, it seems that it should be possible to build ESM version on our own.

The benefit would be as previously mentioned, i.e. there would be no room for conflict between MathJax loaded by us and MathJax present on the web page or loaded by other libraries. We could also serve this from our CDN.

Reply:
Yeah, we can build our won version with ESM, but how do I serve it from our CDN? I will work on the script

Reply:
We control the contents of `cdn.bokeh.org`